packages/core/src/api/user.ts

```diff
@@ -57,43 +61,51 @@ export class UsersAPI {
 	 *
 	 * @see {@link https://discord.com/developers/docs/resources/user#leave-guild}
 	 * @param guildId - The id of the guild
+	 * @param options - The options for leaving the guild
 	 */
-	public async leaveGuild(guildId: Snowflake) {
-		await this.rest.delete(Routes.userGuild(guildId));
+	public async leaveGuild(guildId: Snowflake, { signal }: Pick<RequestData, 'signal'> = {}) {
+		await this.rest.delete(Routes.userGuild(guildId), { signal });
 	}
 
 	/**
 	 * Edits the current user
 	 *
 	 * @see {@link https://discord.com/developers/docs/resources/user#modify-current-user}
-	 * @param user - The new data for the current user
+	 * @param body - The new data for the current user
+	 * @param options - The options for editing the user
 	 */
-	public async edit(user: RESTPatchAPICurrentUserJSONBody) {
-		return this.rest.patch(Routes.user('@me'), { body: user }) as Promise<RESTPatchAPICurrentUserResult>;
+	public async edit(body: RESTPatchAPICurrentUserJSONBody, { signal }: Pick<RequestData, 'signal'> = {}) {
+		return this.rest.patch(Routes.user('@me'), { body, signal }) as Promise<RESTPatchAPICurrentUserResult>;
 	}
 
 	/**
 	 * Fetches the guild member for the current user
 	 *
 	 * @see {@link https://discord.com/developers/docs/resources/user#get-current-user-guild-member}
 	 * @param guildId - The id of the guild
+	 * @param options - The options for fetching the guild member
 	 */
-	public async getGuildMember(guildId: Snowflake) {
-		return this.rest.get(Routes.userGuildMember(guildId)) as Promise<RESTGetCurrentUserGuildMemberResult>;
+	public async getGuildMember(guildId: Snowflake, { signal }: Pick<RequestData, 'signal'> = {}) {
+		return this.rest.get(Routes.userGuildMember(guildId), { signal }) as Promise<RESTGetCurrentUserGuildMemberResult>;
 	}
 
 	/**
 	 * Edits the guild member for the current user
 	 *
 	 * @see {@link https://discord.com/developers/docs/resources/guild#modify-current-member}
 	 * @param guildId - The id of the guild
-	 * @param member - The new data for the guild member
-	 * @param reason - The reason for editing this guild member
+	 * @param body - The new data for the guild member
+	 * @param options - The options for editing the guild member
 	 */
-	public async editGuildMember(guildId: Snowflake, member: RESTPatchAPIGuildMemberJSONBody = {}, reason?: string) {
+	public async editGuildMember(
```

Comment:
This fits better under guild as `editCurrentGuildMember`, the same reason for the voice state one 👀